ocp/jenkins.sh

```diff
@@ -11,16 +11,24 @@ fi
 oc import-image jenkins-2-rhel7 --from=registry.access.redhat.com/openshift3/jenkins-2-rhel7:v3.11.82-4 --confirm
 
 ## Customize the the image imported above with all the build tools we need
-oc new-build -D $'FROM jenkins-2-rhel7:latest \n
-      USER root\nRUN yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm && yum install -y python-setuptools puppet rubygem-puppet-lint && yum clean all && easy_install pip && pip install yamllint\n
-      USER 1001' --name=puppet-jenkins
+oc new-build -D $'FROM jenkins-2-rhel7:latest\n
+      USER root\n
+      RUN rpm --import https://yum.puppetlabs.com/RPM-GPG-KEY-puppet && yum-config-manager --add-repo https://yum.puppet.com/puppet5/el/7/x86_64/ && yum -y install puppet-agent && yum install -y https://dl.fedoraproject.org/pub/epel/epel-release-latest-7.noarch.rpm && yum install -y python-setuptools rubygem-puppet-lint gcc zlib-devel gcc-c++ && yum install -y http://mirror.centos.org/centos/7/updates/x86_64/Packages/ruby-devel-2.0.0.648-34.el7_6.x86_64.rpm && yum clean all && easy_install pip && pip install yamllint\n
```

Comment:
can you use a redhat ruby package for this?. If doing yum install $package, it should take all dependencies automatically, obsoleting installing gcc etc..

Reply:
done, gcc zlib-devel gcc-c++ still required